app/api/events/$id.mjs

```diff
@@ -1,9 +1,38 @@
-import { events } from '../events.mjs'
+import events from '../../data/events.json' assert { type: 'json' }
+import sponsors from '../../data/sponsors.json' assert { type: 'json' }
+import talks from '../../data/talks.json' assert { type: 'json' }
+import speakers from '../../data/speakers.json' assert { type: 'json' }
+
+function inflateTalk(talk_id) {
+  let talk = talks.find(t => t.id === talk_id)
+  if (talk && talk.speaker_id) {
+    talk.speaker = speakers.find(s => s.id === talk.speaker_id)
+  }
+  return talk
+}
+
+function inflateSponsors(sponsor_id) {
+  return sponsors.find(s => s.id === sponsor_id)
+}
+
+function inflateEvent(event) {
+  if (event.talks) {
+    event.talks = event.talks.map(inflateTalk)
+    event.sponsors = event.sponsors.map(inflateSponsors)
+  }
+  return event
+}
 
 export async function get(req) {
-  let { id } = req.params
-  let event = events.find(s => s.id === id)
+  let inflatedEvents = events.map(inflateEvent)
+  let event = [inflatedEvents.find(e => e.id === req.params.id)]
+  let eventSponsors = event[0].sponsors
+  let eventTalks = event[0].talks
   return {
-    json: { event }
+    json: {
+      events: event,
+      sponsors: eventSponsors,
+      talks: eventTalks
+    }
```

Comment:
Future me is totally gonna forget that we are returning this particular contract to reuse the list-talks, etc elements on this page. Can you add a comment here to that effect?

I'm definitely gonna come back later and be like "why are we returning a list here?" and try to change that and break everything 😆

Reply:
✅